src/emitter/emitter.ts

```diff
@@ -35,6 +49,39 @@ export class EventEmitter<E extends object> implements IEventEmitter<E> {
   public dispose() {
     this._emitter.removeAllListeners();
   }
+
+  public createEmitGroup(): {
+    queueCount: number;
+    emit: EventEmitter<E>['emit'];
+    flush: () => void;
+    reset: () => void;
+  } {
+    const notifier = this;
+
+    let eventQueue: Array<[unknown, Array<unknown>]> = [];
+
+    return {
+      get queueCount() {
+        return eventQueue.length;
+      },
+
+      emit(event: any, ...args: any) {
+        eventQueue.push([event, args]);
+      },
+
+      flush() {
+        for (const [event, args] of eventQueue) {
+          notifier.emit(event, ...args);
+        }
+
+        this.reset();
+      },
+
+      reset() {
+        eventQueue = [];
```

Comment:
```suggestion
        eventQueue.length = 0;
```
I usually feel safer when we actually resize the same array, rather than allocating a new one.
We are not leaking `eventQueue` now, so it doesn't really matter, but if we ever decided to add a method exposing `eventQueue` such as `getQueue`, looking more or less like this
```ts
getQueue: () => eventQueue
``` 
we'd be most likely bit.
To illustrate
```ts
const emitGroup = emitter.createEmitGroup();
const queue = emitter.getQueue();
emitGroup.emit('foo');
emitGroup.emit('bar');
queue; // [['foo', []], ['bar', []]]
emitGroup.queueCount; // 2
emitGroup.flush();
queue; // [['foo', []], ['bar', []]]
emitGroup.queueCount; // 0
```